lib/teleterm/clusters/cluster_servers.go

```diff
@@ -68,3 +71,80 @@ func (c *Cluster) GetServers(ctx context.Context) ([]Server, error) {
 
 	return results, nil
 }
+
+func (c *Cluster) GetServers(ctx context.Context, r *api.GetServersRequest) (*GetServersResponse, error) {
+	var (
+		clusterServers []types.Server
+		resp           *types.ListResourcesResponse
+		sortBy         types.SortBy
+	)
+
+	err := addMetadataToRetryableError(ctx, func() error {
+		proxyClient, err := c.clusterClient.ConnectToProxy(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer proxyClient.Close()
+
+		authClient, err := proxyClient.CurrentClusterAccessPoint(ctx)
+		// do we need to call authClient.Close() similar to proxyClient above?
+		// is there a better way to get the auth client instead of this?
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		sortParam := r.SortBy
+		if sortParam != "" {
+			vals := strings.Split(sortParam, ":")
+			if vals[0] != "" {
+				sortBy.Field = vals[0]
+				if len(vals) > 1 && vals[1] == "desc" {
+					sortBy.IsDesc = true
+				}
+			}
+		}
```

Comment:
If this was copied from somewhere else in the project, let us know from where and we'll find a good spot for it where it can be shared across Connect and tsh or the web backend.

Reply:
It exists here
https://github.com/gravitational/teleport/blob/7204503cca9223740d3f5868ccdfacb32fd2b1d3/lib/web/resources.go#L345-L355

Reply:
I looked up how `types.SortBy` is used within `api/types`. It seems that we could create `api/types/sortby.go` and add there a single function which converts a string to `SortBy`? I'm just not sure what Go's conventions are in terms of naming functions like that.

Reply:
I've seen something similar [here](https://github.com/gravitational/teleport/blob/f53f3e77d61025d1bd0c8f3fdaa8e3521651a4b5/lib/web/resources.go#L337) but it's defined in `apiserver.go`. I like your idea and I'll move to api/types/sortby.go I suppose. I have no real horse in the race so whatever works is fine with me

Reply:
I moved the sortBy into its own method at `api/types/sortby`. I didn't refactor any of the webapi code to use this yet as I think it would be weird to refactor something unrelated in the scope of this but, we can handle that later. I will reuse this method for the upcoming change to server side database/apps/kubes/ etc etc for Connect in the future